PlagiarismPlugin.inc.php

```diff
@@ -221,37 +512,150 @@ function manage($args, $request) {
                 }
                 return new JSONMessage(true, $form->fetch($request));
         }
+
         return parent::manage($args, $request);
     }
-}
 
-/**
- * Low-budget mock class for \bsobbe\ithenticate\Ithenticate -- Replace the
- * constructor above with this class name to log API usage instead of
- * interacting with the iThenticate service.
- */
-class TestIthenticate {
-	public function __construct($username, $password) {
-		error_log("Constructing iThenticate: $username $password");
+	/**
+	 * Get the cached EULA details form Context
+	 * 
+	 * @param Context 		$context
+	 * @param string|null 	$locale
+	 * 
+	 * @return array
+	 */
+	public function getContextEulaDetails($context, $locale = null) {
+		/** @var \FileCache $cache */
+		$cache = CacheManager::getManager()
+			->getCache(
+				'ithenticate_eula', 
+				"CONTEXT_ID_{$context->getId()}",
```

Comment:
Just use the numeric context ID without the `CONTEXT_ID_`...  prefix -- unless you expect to mix different kinds of IDs into the `ithenticate_eula` cache. (Here and below)